internal/graph/cursors_test.go

```diff
@@ -212,3 +213,99 @@ func TestCombineCursorsWithNil(t *testing.T) {
 	require.Equal(t, combined.AtRevision, revision.String())
 	require.Equal(t, []string{"d", "e", "f"}, combined.Sections)
 }
+
+func TestWithParallelizedStreamingIterableInCursor(t *testing.T) {
+	limits, _ := newLimitTracker(context.Background(), 50)
+	revision := revision.NewFromDecimal(decimal.NewFromInt(1))
+
+	ci, err := newCursorInformation(&v1.Cursor{
+		AtRevision: revision.String(),
+		Sections:   []string{},
+	}, revision, limits)
+	require.NoError(t, err)
+
+	items := []int{10, 20, 30, 40, 50}
+	parentStream := dispatch.NewCollectingDispatchStream[int](context.Background())
+	err = withParallelizedStreamingIterableInCursor[int, int](
+		context.Background(),
+		ci,
+		"iter",
+		items,
+		parentStream,
+		2,
+		func(ctx context.Context, cc cursorInformation, item int, stream dispatch.Stream[int]) error {
```

Comment:
nothing in the tests is currently asserting that there actually is concurrency compared to the old implementation

what about recording start/stop times for each runner and asserting overlap? or have the runners increment a shared counter on entry, decrement on exit, and assert the high watermark of 2?

Reply:
I don't think we need to assert parallelism; so long as it is correct, the parallelism is just a bonus for performance

Reply:
sure, but behavior-wise there's really nothing to distinguish the changes in this PR from the existing implementation. maybe a benchmark?

Reply:
if you've tested manually separate that seems fine. but it seemed easy to just add an atomic Uint and track a high water mark of concurrent processes (that would confirm concurrency and the the concurrency limit at the same time)

Reply:
I added a test that ensures parallelism